zsh-window-title.zsh

```diff
@@ -4,139 +4,182 @@
-	'builtin' 'typeset' -gir __zsh_window_title_debug_default
+    __zsh_window_title_debug_default=0 && \
+    'builtin' 'typeset' -gir __zsh_window_title_debug_default
 
 'builtin' 'typeset' -gi +r __zsh_window_title_directory_depth_default >/dev/null && \
-	__zsh_window_title_directory_depth_default=2 && \
-	'builtin' 'typeset' -gir __zsh_window_title_directory_depth_default
+    __zsh_window_title_directory_depth_default=2 && \
+    'builtin' 'typeset' -gir __zsh_window_title_directory_depth_default
 
 'builtin' 'typeset' -gi +r __zwt_debug_default >/dev/null && \
-	__zwt_debug_default=0 && \
-	'builtin' 'typeset' -gir __zwt_debug_default
-
+    __zwt_debug_default=0 && \
+    'builtin' 'typeset' -gir __zwt_debug_default
 
 # zwt CLI subcommands
 
 __zwt:debugger() {
-	'builtin' 'emulate' -LR zsh
+    'builtin' 'emulate' -LR zsh
 
-	(( ZWT_DEBUG )) && 'builtin' 'print' $funcstack[2]
+    (( ZWT_DEBUG )) && 'builtin' 'print' $funcstack[2]
 }
 
 __zwt:help() {
-	'builtin' 'emulate' -LR zsh
-	__zwt:debugger
+    'builtin' 'emulate' -LR zsh
+    __zwt:debugger
 
-	'command' 'man' zwt 2>/dev/null || 'command' 'man' $__zwt_dir/man/man1/zwt.1
+    'command' 'man' zwt 2>/dev/null || 'command' 'man' $__zwt_dir/man/man1/zwt.1
 }
 
 __zwt:restore-defaults() {
-	'builtin' 'emulate' -LR zsh
-	__zwt:debugger
+    'builtin' 'emulate' -LR zsh
+    __zwt:debugger
 
-	ZSH_WINDOW_TITLE_DEBUG=$__zsh_window_title_debug_default
-	ZSH_WINDOW_TITLE_DIRECTORY_DEPTH=$__zsh_window_title_directory_depth_default
-	ZWT_DEBUG=$__zwt_debug_default
+    ZSH_WINDOW_TITLE_DEBUG=$__zsh_window_title_debug_default
+    ZSH_WINDOW_TITLE_DIRECTORY_DEPTH=$__zsh_window_title_directory_depth_default
+    ZWT_DEBUG=$__zwt_debug_default
 }
 
 __zwt:version() {
-	'builtin' 'emulate' -LR zsh
-	__zwt:debugger
+    'builtin' 'emulate' -LR zsh
+    __zwt:debugger
 
-	'builtin' 'print' zwt $ZWT_VERSION
+    'builtin' 'print' zwt $ZWT_VERSION
 }
 
-
 # zsh-window-title subcommands
 
 __zsh-window-title:debugger() {
-	'builtin' 'emulate' -LR zsh
+    'builtin' 'emulate' -LR zsh
 
-	(( ZSH_WINDOW_TITLE_DEBUG )) && 'builtin' 'print' $funcstack[2]
+    (( ZSH_WINDOW_TITLE_DEBUG )) && 'builtin' 'print' $funcstack[2]
 }
 
 __zsh-window-title:add-hooks() {
-	'builtin' 'emulate' -LR zsh
-	__zsh-window-title:debugger
+    'builtin' 'emulate' -LR zsh
+    __zsh-window-title:debugger
 
-	# update window title before drawing the prompt
-	add-zsh-hook precmd __zsh-window-title:precmd
+    # update window title before drawing the prompt
+    add-zsh-hook precmd __zsh-window-title:precmd
+    add-zsh-hook precmd __zsh-window-title:precmd-update  # New hook for updating title after command
 
-	# update the window title before executing a command
-	add-zsh-hook preexec __zsh-window-title:preexec
+    # update the window title before executing a command
+    add-zsh-hook preexec __zsh-window-title:preexec
+    add-zsh-hook preexec __zsh-window-title:preexec-update  # New hook for updating title before execution
```

Comment:
- What's the motivation for adding a additional hooks instead of modifying __zsh-window-title:precmd and/or add-zsh-hook preexec __zsh-window-title:preexec?
- The "New …" comments won't age well

Reply:
Reverted this!